cdap-ui/app/directives/start-stop-button/start-stop-button.js

```diff
@@ -0,0 +1,78 @@
+angular.module(PKG.name + '.commons')
+  .directive('myStartStopButton', function() {
+    return {
+      restrict: 'E',
+      scope: {
+        type: '@',
+        isStoppable: '@',
+        isPreferences: '@',
+        preferencesHandler: '&',
+        runtimeHandler: '&'
+      },
+      templateUrl: 'start-stop-button/start-stop-button.html',
+      controller: function($scope, $state, MyDataSource, myRuntimeService, myProgramPreferencesService) {
+        $scope.isStoppable = ($scope.isStoppable == 'true');
+        
+        if ( typeof $scope.isPreferences === 'undefined') {
+          $scope.isPreferences = true;
+        } else {
+          $scope.isPreferences = !!$scope.isPreferences;
+        }
+        $scope.runtimeArgs = [];
+        var path = '/apps/' + $state.params.appId +
+                   '/' + $scope.type + '/' + $state.params.programId;
+        var dataSrc = new MyDataSource($scope);
+
+        // Poll for status
+        dataSrc.poll({
+          _cdapNsPath: path + '/status'
+        }, function(res) {
+          $scope.status = res.status;
+          console.info($scope.isStoppable, $scope.status);
+        });
+
+        // Do 'action'. (start/stop)
+        $scope.do = function(action) {
+          var requestObj = {};
+          requestObj = {
+            _cdapNsPath: path + '/' + action,
+            method: 'POST'
+          };
+          if (action === 'start') {
+            $scope.status = 'STARTING';
+            if ($scope.runtimeArgs && Object.keys($scope.runtimeArgs).length > 0) {
```

Comment:
since you are explicitly instantiating $scope.runtimeArgs on line 21, do we need the first condition? It will always be true, right?